src/btree/bt_read.c

```diff
@@ -282,6 +282,9 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
     if (!LF_ISSET(WT_READ_CACHE))
         WT_STAT_CONN_DATA_INCR(session, cache_pages_requested);
 
+    /* Encourage a race */
+    __wt_timing_stress(session, WT_TIMING_STRESS_AGGRESSIVE_STASH_FREE, NULL);
```

Comment:
I don't think this is quite right. It's going to put a sleep into this code path, but we want to call `wt_stash_discard`. I think it would look something like:

```
if (FLD_ISSET(S2C(session)->timing_stress_flags, WT_TIMING_STRESS_AGGRESSIVE_STASH_FREE))
    __wt_stash_discard(session);
```

Reply:
Sorry... I don't know why I ended up adding `__wt_timing_stress` while we discussed `__wt_stash_discard` the whole time. I will fix it, thanks for noticing it.

Reply:
Updated in e66645c5f72d1ec0a17dee9e52fa12dc83a9a9ba